### PR TITLE
Fix more deprecation warnings

### PR DESCRIPTION
**Pull request by jamesstout**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Fixes more deprecation warnings

Any other comments?
-------------------
Tested most of it ... 

Where has this been tested?
---------------------------
**Devices/Simulators:** iMac19,1

**macOS Version:**: 10.15.7 (19H2)

**Sequel-Ace Version:** main-3_0_0 latest

### CODE REVIEW

File: Source/Other/Extensions/StringExtension.swift [hunk 1]
```diff
@@ -41,6 +41,24 @@ extension String {
...
+	// we should check the string first
+	var isPercentEncoded: Bool {
+		let decoded = self.removingPercentEncoding
+		
+		if (decoded == nil) || (decoded == self) {
+			return false
+		}
+		else {
+			return true
+		}
```
Comment (Jason-Morcos) [inline Source/Other/Extensions/StringExtension.swift:59]: Could turn into one line: `return (decoded != nil) && (decoded != self);`

**Review by Jason-Morcos — APPROVED** [review]

File: Source/Other/Extensions/StringExtension.swift (hunk 1 above)
Comment (Kaspik) [reply Source/Other/Extensions/StringExtension.swift:59]: Better to use ´guard let decoded = self.stringByReplacingPercentEncoding else { return false }´
Then ´return self != decoded´

**Review by Kaspik — APPROVED** [review]

File: Source/Other/Extensions/StringExtension.swift (hunk 1 above)
Comment (Jason-Morcos) [reply Source/Other/Extensions/StringExtension.swift:59]: Oooo even better!

Comment (jamesstout) [reply Source/Other/Extensions/StringExtension.swift:59]: will update

**Review by Kaspik — APPROVED** [review]

**Pull request merged**